src/components/20-molecules/dropdown/index.js

```diff
@@ -251,14 +256,62 @@ class AXADropdown extends NoShadowDOM {
         // case 2: already open, navigate
         this.navigateByKeyboard(arrowKey);
       }
+    } else {
+      // selection of first matching dropdown item by quickly typing a prefix
+      // of the item's 'name' value, e.g. 'afgh' for selecting Afghanistan in a country list
+      this.handleAutosuggest(key);
+    }
+  }
+
+  handleAutosuggest(key) {
+    // ready for autosuggestions, by having focus on the non-native UI?
+    if (document.activeElement !== this.button) {
+      // no, bail out
+      return;
     }
+    // do we have even a chance of matching for the key that was just pressed?
+    const character = key.toLowerCase();
+    if (!this._autosuggestLetters.has(character)) {
+      // no, bail out
+      return;
+    }
+
+    this._typedSoFar += character;
+    // wait for certain period of keyboard inactivitity...
+    clearTimeout(this._timer);
+    this._timer = setTimeout(() => {
+      // ... before trying to match what was typed sofar against items
+      const { _autosuggestDictionary, _typedSoFar, items } = this;
+      const match = _autosuggestDictionary.indexOf(INDEX_END + _typedSoFar);
+      if (match > -1) {
+        // we have a match, determine the corresponding item ...
+        const matchIndexPos =
+          _autosuggestDictionary.indexOf(WORD_END, match) + 1;
+        const firstSelectedItem = parseInt(
+          _autosuggestDictionary.slice(matchIndexPos),
+          10
+        );
+        // ... and its value
+        const { value } = items[firstSelectedItem];
+        // re-render to visually select matched item
+        this.updateItems(value, 'scrollIntoView');
+      }
+      // next characters start a *new* autosuggestion
+      this._typedSoFar = '';
+    }, AUTOSUGGEST_INACTIVITY_DELAY);
   }
 
   handleDropdownClick(e) {
     e.preventDefault();
     e.stopPropagation();
     // toggle dropdown
-    this.openDropdown(!this.open);
+    const { open } = this;
+    this.openDropdown(!open);
+    if (open) {
+      // note: without setTimeout, scrolling is off by several items, s.t. the selected
+      // element is not visible (at least with maxHeight being set)
+      setTimeout(() => this.scrollIntoView(), 1);
```

Comment:
Why 1 in the end and not leaving it out?

Reply:
uh, timeout needs a ... time argument.